ang/civiawards/shared/directives/contacts-group-dropdown.directive.js

```diff
@@ -0,0 +1,193 @@
+(function (angular, $, _) {
+  var module = angular.module('civiawards');
+
+  module.directive('civiawardContactsGroupDropdown', function (crmApi) {
+    return {
+      link: civiawardContactsGroupDropdownLink,
+      restrict: 'A',
+      require: 'ngModel'
+    };
+
+    /**
+     * Contacts Group Dropdown Link function
+     *
+     * @param {object} scope scope object of the controller
+     * @param {object} elem directives element
+     * @param {object} attr attributes of the directive element
+     * @param {object} ngModel ng model service
+     */
+    function civiawardContactsGroupDropdownLink (scope, elem, attr, ngModel) {
+      scope.groups = {
+        more: false,
+        results: [{
+          text: 'Include Group',
+          children: []
+        }, {
+          text: 'Exclude Group',
+          children: []
+        }]
+      };
+
+      (function init () {
+        ngModel.$render = refreshUI;
+
+        fetchGroupsData()
+          .then(prepareGroupsDataForSelect2)
+          .then(initSelect2)
+          .then(refreshUI);
+
+        scope.$watchCollection('recips.include', refreshUI);
+        scope.$watchCollection('recips.exclude', refreshUI);
```

Comment:
is it possible to avoid using watch at all? Like adding an `ng-change` or similar to the view? Also, we don't need to watch for a specific inclusion/exclusion, it should trigger the refresh on any change.